acceptance/tests/validate_vendored_ruby.rb

```diff
@@ -146,12 +146,26 @@ def install_dependencies(agent)
     end
   end
 
-  step "'irb' successfully loads gems" do
+  step "vendored ruby successfully loads gems" do
     agents.each do |agent|
       irb = irb_command(agent)
+
       ['hocon', 'deep_merge', 'puppet/resource_api'].each do |gem|
-        stdout = on(agent, "echo \"require '#{gem}'\" | #{irb}").stdout.chomp
-        assert_match(/true/, stdout, "'irb' failed to require the #{gem} gem")
+        if agent['platform'] =~ /windows*/
+          # Although beaker can set a PATH for puppet-agent on Windows, it isn't
+          # able to set up the environment for the purpose of loading gems this
+          # way; See BKR-1445 and BKR-986. As a workaround, run the environment
+          # script before requiring the gem. We're using `ruby -r` instead of
+          # irb here because `cmd /c` will only run one argument command before
+          # returning to cygwin (additional arguments will be treated as bash)
+          # and using irb introduces too many quotes.
+          cmd = "env PATH=\"#{agent['privatebindir']}:${PATH}\" cmd /c \"environment.bat && ruby -r#{gem} -e 'puts true'\""
```

Comment:
I would expect the `environment.bat` modifications to the LOAD_PATH/RUBYLIB to only affect the `cmd.exe` process/interpreter, but not the `ruby` process that cygwin invokes? If it doesn't work, you may need to create a wrapper batch script that sources `environment.bat` and then invokes ruby:

```bat
call environment.bat
ruby -r #{gem} -e 'puts true'
```
And then invoke wrapper script using:

```bash
cmd = "env PATH=\"#{agent['privatebindir']}:${PATH}\" cmd /c wrapper.bat"
```

Reply:
oh, I see that calling `environment.bat` and `ruby` are within the double quoted string passed to `cmd`, so that makes sense. 👍